### ISSUE TITLE
Can't login to extension - 2fa box empty

### ISSUE THREAD

**Issue by ShizCalev** [bitwarden/clients#2426]
### Steps To Reproduce

Fresh install of Firefox on a fresh install of andriod
1) go to login
2) put in username and password
3) press login

### Expected Result

See a 2fa box so you can finish logging in

### Actual Result

The 2fa box is empty. 

### Screenshots or Videos

![Screenshot_20220302-120803_Firefox](https://user-images.githubusercontent.com/6209658/156412838-0d0ed85b-8fa4-4a36-9f93-c24daa17f2ee.png)
![Screenshot_20220302-120806_Firefox](https://user-images.githubusercontent.com/6209658/156412842-ba8b3d77-a651-4639-8d56-4dcf3bf1fcb7.png)

### Operating System

Android

### Operating System Version

12

### Web Browser

Firefox

### Browser Version

97.2.0

### Build Version

can't see it since you have to be logged in to get to the settings screen. latest version as of this being opened.

**Comment by MGibson1** [comment]
Hi, could you navigate to the [captcha prompt hosted by bitwarden](https://vault.bitwarden.com/captcha-connector.html?data=eyJzaXRlS2V5IjoiYmMzOGM4YTItNTMxMS00ZThjLTlkZmMtNDllOTlmNmRmNDE3IiwibG9jYWxlIjoiZW4ifQ%3D%3D&parent=https%253A%252F%252Fvault.bitwarden.com%252F%2523%252F&v=1), directly? Do you still see only a grey box?

What about if you go [directly to hcaptcha](https://newassets.hcaptcha.com/captcha/v1/716b3bf/static/hcaptcha-checkbox.html#id=0l155ps8mcri&host=vault.bitwarden.com&sentry=true&reportapi=https%3A%2F%2Faccounts.hcaptcha.com&recaptchacompat=true&custom=false&tplinks=on&sitekey=bc38c8a2-5311-4e8c-9dfc-49e99f6df417&theme=light)?

Trying to understand where the failure point is.

**Comment by ShizCalev** [comment]
![Screenshot_20220302-164715_Firefox](https://user-images.githubusercontent.com/6209658/156455035-dc00631e-0354-47fe-b063-f7a9ffd399f5.png)
![Screenshot_20220302-164707_Firefox](https://user-images.githubusercontent.com/6209658/156455037-90a6da29-8096-4d86-a965-7eb2da8e27ca.png)
Both of those appear fine.

**Comment by skoshy** [comment]
Happening for me as well; is there a workaround for the moment?

**Comment by MGibson1** [comment]
@skoshy are you also using bitwarden extension in ff on android?

Can you give any specifics about how you're accessing Bitwarden? 
* Are you behind a vpn?
* what browser are you using?
* Is this happening for the other bitwarden clients for you?

I've been having a very hard time reproducing this issue. Any specifics you can share would be very helpful to try and narrow this down

**Comment by skoshy** [comment]
Yup, I'm using Firefox Nightly and on Android 12.

I installed the extension via the menu in Firefox Nightly, and I then tried logging into the extension.

Not behind a VPN, and accessing the links posted above directly going to HCaptcha works fine.

For what it's worth, I also tried using Kiwi Browser, which is Chromium based and supports extensions on Android, and the HCaptcha shows up properly when logging in there.

I can also log into the actual Android Bitwarden app with captcha, and also on my PC.

**Comment by ShizCalev** [comment]
Regular Firefox here, fresh install, no vpn, and the bitwarden app works fine, only the browser extension (1.56.6) is having the issue for me.

**Comment by kspearrin** [comment]
Are any of you able to join our gitter chat [here ](https://gitter.im/bitwarden/Lobby) and help us debug this in live chat together? We are still having a hard time reproducing the issue and your help would be appreciated.

https://gitter.im/bitwarden/Lobby

**Comment by eliykat** [comment]
This also seems to be affecting Vivaldi on PopOS (desktop): #2432

**Comment by Greenderella** [comment]
@eliykat I'm getting the captcha correctly rendered on Pop!_OS 21.10 and Vivaldi 5.1.2567.49, Bitwarden Version: 1.56.6

![image](https://user-images.githubusercontent.com/5418577/156745268-fff43505-12f1-4c0f-8a25-b75a6e71e3e4.png)

**Comment by Mac-lucky** [comment]
Same issue for me with Firefox nightly for Android.

**Comment by fortinj1354** [comment]
I'm seeing the same behavior in a fresh install of the latest stable version of Firefox for Android. 

Firefox 97.2.0, latest version of the Bitwarden extension installed via the app, no other extensions installed and enhanced tracking protection turned fully off

**Comment by MGibson1** [comment]
We're loading the captcha through an iframe for a few reasons. It seems that firefox for android just plain isn't loading iframe for addon popup windows.

We're currently investigating workarounds, but it looks like this will not be a very quick fix. For those of you impacted by this, the fastest workaround will be the mobile application.

**Comment by ShizCalev** [comment]
It's suddenly working for me now, nothing has changed on my end. 
(Still ff 97.2.0, bw 1.56.6)

**Comment by jayres92** [comment]
If it helps anyone here I figured a way around this to help me log into the extension.

My environment:

OS: Windows 11
Browser: LibreWolf v97.0.2-1 (Latest)
Extensions: BitWarden 1.56.6 and uBlock Origin 1.41.8

How I got around to use the extension.

1. If you have uBlock Origin on disable it (or VPN, DNS, or a different adblocker disable it for the time being).

2. Open the two links from @MGibson1 posted above and do the captcha -

> Hi, could you navigate to the [captcha prompt hosted by bitwarden](https://vault.bitwarden.com/captcha-connector.html?data=eyJzaXRlS2V5IjoiYmMzOGM4YTItNTMxMS00ZThjLTlkZmMtNDllOTlmNmRmNDE3IiwibG9jYWxlIjoiZW4ifQ%3D%3D&parent=https%253A%252F%252Fvault.bitwarden.com%252F%2523%252F&v=1), directly? Do you still see only a grey box?
> 
> What about if you go [directly to hcaptcha](https://newassets.hcaptcha.com/captcha/v1/716b3bf/static/hcaptcha-checkbox.html#id=0l155ps8mcri&host=vault.bitwarden.com&sentry=true&reportapi=https%3A%2F%2Faccounts.hcaptcha.com&recaptchacompat=true&custom=false&tplinks=on&sitekey=bc38c8a2-5311-4e8c-9dfc-49e99f6df417&theme=light)?
> 
> Trying to understand where the failure point is.

3. Open a new tab and try to use your BitWarden extension and login now.

4. If you still cannot log in try this in addition (I've done this originally, but thought if my uBlock Origin was off then I wouldn't need these noop rules) - add a permanent noop rule and add a _**temporary**_ noop rule for 3P.
`* hcaptcha.com * noop`
` * newassets.hcaptcha.com * noop`
Temporary noop
`vault.bitwarden.com * 3p noop`
`newassets.hcaptcha.com * 3p noop`

5. If you are able to use the extension and all is well don't forget remove the temporary noop rule for 3P in your uBO extension.

**Comment by dbfmuk** [comment]
I was having this issue on desktop browsers. Both Firefox (97.02) and also Brave (1.36.111). Thought it was strange that it occurred on 2 different browsers

After some investigation, the issue that was causing it for me may have been 'Noscript' add-on/extension blocking hcaptcha.com. I worked around this by going directly to hcaptcha.com within the browser and marking this site as 'trusted' within Noscript. Now when I tried to login to Bitwarden, the captcha was shown.

**Comment by jurdanetac** [comment]
@dbfmuk workaround fixed it for me on regular Firefox (98.0) for desktop.

**Comment by OfficialMuffin** [comment]
Same issue here on latest Brave browser on Windows 10

**Comment by hlovdal** [comment]
Same issue here, Firefox 97.0.1 on Linux desktop (with noscript, ublock origin, privacy badger, canvas blocker (and some more addons)).

Trying to log in from the browser extension icon does not work, nothing is shown in the white space that supposed to contain captcha content (although there is a short flicker where a scrollbar flashes).

If I open a new tab and go to https://vault.bitwarden.com/ I am able log in there after I add <https://hcaptcha.com> and <https://newassets.hcaptcha.com> as trusted.

This however does not affect the browser icon login, it is still failing after configuring hcaptcha as trusted.

**Comment by shaenr** [comment]
Same issue. 

The browser extension tries to load what appears to be a captcha or some form of 2FA after entering login credentials and master password, but nothing loads in the space that appears. The icon has become greyed out (not longer blue) as well. Have tried in private and normal session, with other extensions turned off, and I have no problem logging in from the web at vault.bitwarden.com, my android phone, or my Windows PC.

Problem with Firefox 98.0.1 (64-bit) on Linux desktop. Bitwarden browser extention for firefox 1.56.6.

**Comment by lisansas** [comment]
> If it helps anyone here I figured a way around this to help me log into the extension.
> 
> My environment:
> 
> OS: Windows 11 Browser: LibreWolf v97.0.2-1 (Latest) Extensions: BitWarden 1.56.6 and uBlock Origin 1.41.8
> 
> How I got around to use the extension.
> 
> 1. If you have uBlock Origin on disable it (or VPN, DNS, or a different adblocker disable it for the time being).
> 2. Open the two links from @MGibson1 posted above and do the captcha -
> 
> > Hi, could you navigate to the [captcha prompt hosted by bitwarden](https://vault.bitwarden.com/captcha-connector.html?data=eyJzaXRlS2V5IjoiYmMzOGM4YTItNTMxMS00ZThjLTlkZmMtNDllOTlmNmRmNDE3IiwibG9jYWxlIjoiZW4ifQ%3D%3D&parent=https%253A%252F%252Fvault.bitwarden.com%252F%2523%252F&v=1), directly? Do you still see only a grey box?
> > What about if you go [directly to hcaptcha](https://newassets.hcaptcha.com/captcha/v1/716b3bf/static/hcaptcha-checkbox.html#id=0l155ps8mcri&host=vault.bitwarden.com&sentry=true&reportapi=https%3A%2F%2Faccounts.hcaptcha.com&recaptchacompat=true&custom=false&tplinks=on&sitekey=bc38c8a2-5311-4e8c-9dfc-49e99f6df417&theme=light)?
> > Trying to understand where the failure point is.
> 
> 3. Open a new tab and try to use your BitWarden extension and login now.
> 4. If you still cannot log in try this in addition (I've done this originally, but thought if my uBlock Origin was off then I wouldn't need these noop rules) - add a permanent noop rule and add a _**temporary**_ noop rule for 3P.
>    `* hcaptcha.com * noop`
>    ` * newassets.hcaptcha.com * noop`
>    Temporary noop
>    `vault.bitwarden.com * 3p noop`
>    `newassets.hcaptcha.com * 3p noop`
> 5. If you are able to use the extension and all is well don't forget remove the temporary noop rule for 3P in your uBO extension.

it works，thanks

**Comment by creallfluharty** [comment]
Any updates on this? I'm currently experiencing the same issue on a fresh android install with no extensions other than bitwarden.

**Comment by MGibson1** [comment]
This is an issue with Firefox extensions not loading iframes properly. The bug I filed with the mozilla android team is here: https://github.com/mozilla-mobile/fenix/issues/24118.

No movement by them as of yet.

**Comment by ComfortableEast** [comment]
> Any updates on this? I'm currently experiencing the same issue on a fresh android install with no extensions other than bitwarden.

Hi,
I was able to solve the issue by logging in at the[ Bitwarden website](https://vault.bitwarden.com/#/login) after turning off uBlock Origin. This shows the captcha there, and you can log in.
Following this, it prompted me to verify my email in the top-right corner. Once your email has been verified, it won't ever ask for a captcha again, be it a Chromium extension (on PC or mobile) or Mozilla add-on (PC or mobile) or even their Android/IOS app. 

TL;DR **Verify your Bitwarden e-mail**

IMO users should be prompted to verify email from wherever they are trying to log in from.

**Comment by tosiara** [comment]
Fresh Android, fresh Firefox - and login is not working. What a surprise after years of using Bitwarden smoothly :/

Is there a way to show captcha without an iframe?

**Comment by chozabu** [comment]
> TL;DR **Verify your Bitwarden e-mail**
> 
> IMO users should be prompted to verify email from wherever they are trying to log in from.

This workaround worked for me but...
I had trouble verifying email, the email would arrive, but clicking opened a page with a message in the top right letting me know there was an error verify email, and to try again.
After several retrys, and disabling ublock origin - I was able to verify email, and long into android client

TL;DR **disable ublock origin? then Verify your Bitwarden e-mail**

**Comment by vcfvct** [comment]
> Fresh Android, fresh Firefox - and login is not working. What a surprise after years of using Bitwarden smoothly :/
> 
> Is there a way to show captcha without an iframe?

Same here. Have been using FF at Android for years. Tried on android tablet(galaxy tab s8), pixel 5(android 13), Oneplus 10T(Android 12), and an old tablet with Android 7, none of them had the Bitwarden worked within FF/FF-nightly. So disappointed. Having uBlock Origin or not does not affect anything.

**Comment by zjays** [comment]
> > Any updates on this? I'm currently experiencing the same issue on a fresh android install with no extensions other than bitwarden.
> 
> Hi, I was able to solve the issue by logging in at the[ Bitwarden website](https://vault.bitwarden.com/#/login) after turning off uBlock Origin. This shows the captcha there, and you can log in. Following this, it prompted me to verify my email in the top-right corner. Once your email has been verified, it won't ever ask for a captcha again, be it a Chromium extension (on PC or mobile) or Mozilla add-on (PC or mobile) or even their Android/IOS app.
> 
> TL;DR **Verify your Bitwarden e-mail**
> 
> IMO users should be prompted to verify email from wherever they are trying to log in from.

Thanks, this worked for me. I logged into the Bitwarden link above on an unused desktop Edge browser and I used the verify my email link from there. If it matters, I opened the verification link sent to my email in the Android Firefox app that I wanted to use Bitwarden on. After that, I was able to login to the Bitwarden app on my newly installed Android Firefox app.

**Comment by vcfvct** [comment]
> 

This is great detailed steps. Thanks. 👍
It finally worked. Still hope the captcha can work in the normal flow.  😄

**Comment by MintMain21** [comment]
I just don't understand why a Captcha is even necessary. Basic 2FA should work enough. I tested it on Desktop Edge after getting frustrated with being unable to log in on Firefox Android and they didn't even ask for a capcha on Desktop Edge.

It's a good thing the Android App works fine, or else we'd be in real hot water. 

Just goes to show that Firefox needs to invest more in Extension support, or at the very least kindly remove Bitwarden from the default list of "supported extensions" until it's properly supported again.

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
